### PR TITLE
add: modal and rating

### PR DESCRIPTION
**Pull request by Shivani-1524**
added two components:
1. Modal
2. Rating

### CODE REVIEW

File: Components/Modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  width: 50%;
+  height: fit-content;
+  background-color: var(--primary-color);
+  padding: 15px;
+}
+
+.modal-btn-wrapper {
+  display: flex;
+  justify-content: flex-end;
+  gap: 20px;
```
Comment (rahulk31) [inline Components/Modal/modal.css]: relative values can be used.

File: Components/Modal/modal.css [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.modal-title i {
+  position: absolute;
+  top: 0;
+  right: 0;
+  padding: 10px 15px;
```
Comment (rahulk31) [inline Components/Modal/modal.css:38]: here as well, relative values can be used.

File: Components/Modal/modal.css [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  color: var(--white);
+}
+
+.modal-container {
+  position: relative;
+  width: 50%;
+  height: fit-content;
+  background-color: var(--primary-color);
+  padding: 15px;
+}
```
Comment (rahulk31) [inline Components/Modal/modal.css:26]: relative values can be used.

File: Components/Modal/modal.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+modalBtnDemo.addEventListener("click", () => {
+    modalWrapper.setAttribute("style", "display: flex");
+});
+
+modalBtnCancel.addEventListener("click", () => {
+    removeModal();
+});
+
+modalBtnClose.addEventListener("click", () => {
+    console.log("BRUH")
```
Comment (rahulk31) [inline Components/Modal/modal.js]: Remove the console statement.

File: Components/Rating/rating.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+//     colorStars(1);
+// });
+
+// // 2 Star Rating
+// ratingStar2.addEventListener("click", () => {
+//     colorStars(2);
+// });
+
+// // 3 Star Rating
+// ratingStar3.addEventListener("click", () => {
```
Comment (rahulk31) [inline Components/Rating/rating.js]: commented code can be removed.

**Review by rahulk31 — APPROVED** [review]
Overall looks good to me.

File: Components/Modal/modal.css [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  align-items: center;
+  justify-content: center;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  width: 100vw;
+  height: 100vh;
+  background-color: rgb(0, 0, 0, 0.7);
+  z-index: 10000;
+  color: var(--white);
```
Comment (Ankur9669) [inline Components/Modal/modal.css:13]: I think z-index is overused😅

**Review by Ankur9669 — APPROVED** [review]
Code looks great to me

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
Can't find any major issues, looks fine to me 👍

**Comment by yaswanthmyneni** [comment]
Everything looks good to me.

File: Components/Modal/modal.css (hunk 6 above)
Comment (Naman-Saxena1) [reply Components/Modal/modal.css:13]: > I think z-index is overused😅

Wow! 😆

File: Components/Modal/modal.css (hunk 1 above)
Comment (Shivani-1524) [reply Components/Modal/modal.css]: didnt want it to scale but will change it!

File: Components/Modal/modal.css (hunk 2 above)
Comment (Shivani-1524) [reply Components/Modal/modal.css:38]: i wouldnt want it to scale which is why its in px

File: Components/Modal/modal.css (hunk 3 above)
Comment (Shivani-1524) [reply Components/Modal/modal.css:26]: i wouldnt want it to scale which is why its in px

File: Components/Modal/modal.js (hunk 4 above)
Comment (Shivani-1524) [reply Components/Modal/modal.js]: Thanks for pointing it out !

File: Components/Rating/rating.js (hunk 5 above)
Comment (Shivani-1524) [reply Components/Rating/rating.js]: okay!

**Comment by Shivani-1524** [comment]
> Code looks great to me

I got stuck in greying stars hence referred to your logic for ratings. Thanks anyway!

**Comment by Shivani-1524** [comment]
> Everything looks good to me.

awesome

**Comment by Shivani-1524** [comment]
> Can't find any major issues, looks fine to me 👍

okay then :)

File: Components/Modal/modal.css (hunk 6 above)
Comment (Shivani-1524) [reply Components/Modal/modal.css:13]: > I think z-index is overused😅

thanks for poining it out!

**Comment by Shivani-1524** [comment]
> Overall looks good to me.

cool :)

**Pull request merged**